src/components/UpgradePlanForm/UpgradePlanForm.js

```diff
@@ -4,45 +4,69 @@ import { Formik, Form, Field, ErrorMessage } from 'formik';
 import { InjectAppServices } from '../../services/pure-di';
 import { Loading } from '../../components/Loading/Loading';
 import { SubmitButton } from '../form-helpers/form-helpers';
+import useTimeout from '../../hooks/useTimeout';
 
 const fieldNames = {
   selectedPlanId: 'selectedPlanId',
   message: 'message',
 };
 /** @type { import('../../services/doppler-legacy-client').DopplerLegacyClient } */
-const UpgradePlanForm = ({ handleClose, isSubscriber, dependencies: { dopplerLegacyClient } }) => {
+const UpgradePlanForm = ({
+  handleClose,
+  isSubscriber,
+  dependencies: { dopplerLegacyClient, appSessionRef },
+}) => {
   const [state, setState] = useState({
     availablePlans: null,
     formIsValid: false,
     isLoading: true,
   });
   const intl = useIntl();
+  const createTimeout = useTimeout();
 
   useEffect(() => {
     const fetchData = async () => {
       const availablePlans = await dopplerLegacyClient.getUpgradePlanData(isSubscriber);
+      const planUser = appSessionRef.current.userData.user.plan;
+      const availablePlansInSameType = availablePlans.filter((p) =>
+        planUser.isSubscribers === true
+          ? p.SubscribersQty > planUser.maxSubscribers
+          : p.EmailQty > planUser.maxSubscribers,
+      );
       setState({
-        availablePlans: availablePlans,
+        availablePlans: availablePlansInSameType,
         isLoading: false,
-        firstPlanId: availablePlans && availablePlans.length && availablePlans[0].IdUserTypePlan,
+        firstPlanId: availablePlansInSameType[0]?.IdUserTypePlan,
       });
       // TODO: what happens when availablePlans is an empty array?
     };
     fetchData();
-  }, [isSubscriber, dopplerLegacyClient]);
+  }, [isSubscriber, dopplerLegacyClient, appSessionRef]);
+
+  const [sentEmail, setSentEmail] = useState(false);
 
   const onSubmit = async (values, { setSubmitting }) => {
-    await dopplerLegacyClient.sendEmailUpgradePlan({
+    await dopplerLegacyClient.upgradePlan({
       IdClientTypePlanSelected: values[fieldNames.selectedPlanId],
       Detail: values[fieldNames.message],
     });
+
     setSubmitting(false);
-    handleClose();
+    if (values[fieldNames.selectedPlanId] > 0) {
+      setSentEmail(false);
+      // TODO: Show a new popup
+    } else {
+      setSentEmail(true);
+      appSessionRef.current.userData.user.isLastPlanRequested = true;
```

Comment:
This value `isLastPlanRequested` I suppose it is updated in `dopplerLegacyClient.upgradePlan`  back end right?

Reply:
Yes in
https://github.com/FromDoppler/doppler-webapp/pull/1479/files#diff-ca1c6aed6deed75a148d78f99ad81b25d02f03d884c6103cc4523e76161567ddR520
and
Doppler Back end here
https://github.com/MakingSense/Doppler/pull/8258/files#diff-4de8340f5a1f9c1870b0ec3950f68cf42e78bda1df42dde1350837630006814eR36